File: backend/src/main/java/com/desafiorh/backend/model/Funcionario.java
package com.desafiorh.backend.model;

import com.desafiorh.backend.enums.StatusFuncionario;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Funcionario {
    private Integer id;
    private String nome;
    private String email;
    private String telefone;
    private String cargo;
    private String departamento;
    private double salario;
    private String cidade;
    private StatusFuncionario status;
}

File: frontend/src/types/Funcionario.ts
export type StatusFuncionario =
  | 'EM_ANALISE'
  | 'APROVADO'
  | 'REPROVADO'
  | 'CONTRATADO'

export interface Funcionario {
  id: number
  nome: string
  email: string
  telefone: string
  cargo: string
  departamento: string
  salario: number
  cidade: string
  status: StatusFuncionario
}

export type FuncionarioRequest = Omit<Funcionario, 'id'>

export type FuncionarioPatchRequest = Partial<FuncionarioRequest>
